### DISCUSSION TITLE
Transparent project/template bootstrapping

### DISCUSSION THREAD

**Discussion by mindplay-dk** [stackblitz/core#1575] [Ideas]
This started as an issue, but it's not really a concrete "please change or fix this" type of request, so I decided to post here instead.

I will tell you my typical use-case, which has always failed, ever since the inception of StackBlitz.

I tend to start with the TypeScript template - and then I want JSX support (and no, I don't want a React project) so I rename the `index.ts` file to `index.tsx` and, shucks, the project is dead.

Confused, I look at the `index.html` file, where (on CodeSandbox, or a local project with a "zero configuration" bundler running locally) I would normally expect to find a `<script src="index.ts">` that I can correct - but no, to my confusion, the HTML file is just a `<div>`.

I then start poking through `package.json` and `tsconfig.json`, the "Info" panel, the "Settings" tab - I frantically search the entire UI for any sort of clue, but still to this day, I have no clue how it defines the entrypoint script, or what a StackBlitz template even is.

Once I've created and named a project, I can't even see anywhere in the UI which template it's using.

The whole experience is very "black box", very confusing, and not really transparent at all.

What I'd like to see, is tab/panel, or at least a file, that provides this information and lets me edit it somehow.

With your ability to run a real container in the browser (which, just, *wow*) perhaps this is something that would be more easily in reach in the future? If each user runs the entire build locally, there's no reason to limit the visibility or editing of any project files, is there?

Hopefully at that point, a template changes from being this mysterious black box - and instead becomes something you configure once, and it bootstraps a proper node project, with all project files visible and editable?

I'm sad to say, but I generally use CodeSandbox instead. I keep coming back to StackBlitz every few months or so though - because it is, in many ways, a much nicer product, and I keep hoping something in this regard.

But I've never been able to test or try anything I wanted to with your templates, none of which are set up for any of the things I like to play with - often something that starts with TypeScript and JSX but none of the mainstream frameworks.

What do you think? Is there any hope of ever evolving this product into something more transparent and flexible? 🙂

**Comment by wernert** [comment]
For my example: i have started via rxjs starter to test out rxjs ideas, 
and then have added another page and TS file for a second test.
I have found now way to wire it up, or to access the new pages.
Regardless of the url it always shows index.html.
Searched the docs and package and project settings, found no hint how to do such a easy thing.

**Reply by Trade-Help** [reply]
**_[]()_**
